activerecord/test/cases/finder_test.rb

```diff
@@ -202,10 +202,15 @@ def test_exists_with_nil_arg
     assert_equal true, Topic.first.replies.exists?
   end
 
-  # Ensure +exists?+ runs without an error by excluding distinct value.
+  # ensures +exists?+ runs without an error by excluding distinct value
   # See https://github.com/rails/rails/pull/26981.
-  def test_exists_with_order_and_distinct
-    assert_equal true, Topic.order(:id).distinct.exists?
+  def test_exists_with_distinct
+    assert_equal true, Topic.distinct.exists?
+  end
+
+  # ensures +exists?+ runs without an error by excluding order value
+  def test_exists_with_order
+    assert_equal true, Topic.order("invalid sql here").exists?
```

Comment:
I agree with your point. Actually I struggled to find out why queries that excluded ORDER BY clause are executed (our company apps has optimized against ORDER BY clause in `relation.to_a`).